buildarr/config/base.py

```diff
@@ -814,49 +814,39 @@ def yaml(
             YAML representation of the model
         """
         return yaml.safe_dump(  # type: ignore[call-overload]
-            json.loads(self.json(*args, **kwargs)),
+            json.loads(self.model_dump_json(*args, **kwargs)),
             **{**(yaml_kwargs or {}), "sort_keys": sort_keys},
         )
 
-    class Config(ModelConfigBase):
-        """
-        Buildarr configuration model class settings.
-
-        Sets some required parameters for serialisation,
-        parsing and validation to work correctly.
-
-        To set additional parameters in your implementing class, subclass this class:
-
-        ```python
-        from __future__ import annotations
+    model_config = model_config_base
+    """
+    Buildarr configuration model configuration.
 
-        from typing import TYPE_CHECKING
-        from buildarr.config import ConfigBase
+    Sets some required parameters for parsing and validation to work correctly.
 
-        if TYPE_CHECKING:
-            from .secrets import ExampleSecrets
+    To set additional parameters in your implementing class, override this attribute:
 
-            class _ExampleConfig(ConfigBase[ExampleSecrets]): ...
-        else:
+    ```python
+    from __future__ import annotations
 
-            class _ExampleConfig(ConfigBase): ...
+    from typing import TYPE_CHECKING
+    from buildarr.config import ConfigBase
 
+    if TYPE_CHECKING:
+        from .secrets import ExampleSecrets
 
-        class ExampleConfig(_ExampleConfig):
-            ...
+        class _ExampleConfig(ConfigBase[ExampleSecrets]): ...
+    else:
 
-            class Config(_ExampleConfig.Config): ...  # Add model configuration attributes here.
-        ```
-        """
+        class _ExampleConfig(ConfigBase): ...
 
-        pass
 
+    class ExampleConfig(_ExampleConfig):
+        ...
 
-# Add custom validators which are not provided by Pydantic.
-_VALIDATORS.extend(
-    [
-        (PurePath, [lambda v: PurePath(v)]),
-        (PurePosixPath, [lambda v: PurePosixPath(v)]),
-        (PureWindowsPath, [lambda v: PureWindowsPath(v)]),
-    ],
-)
```

Comment:
Add unit tests for these types, to make sure they are validated and serialised properly.

Reply:
Unit tests added. Need to add formatters and encoders to the default Buildarr functions, now working properly. Serialisation works without issues, now that we are on Pydantic V2.